templates/gocd-agent-deployment.yaml

```diff
@@ -1,20 +1,25 @@
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: {{ .Values.agent.name }}
+  name: {{ .Values.agent.name }}-deployment
   labels:
-    app: {{ template "name" . }}
+    app: {{ .Values.agent.name }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    role: agent
 spec:
   replicas: {{ .Values.agent.replicaCount }}
   template:
     metadata:
       labels:
-        app: {{ template "name" . }}
+        app: {{ .Values.agent.name }}
+        chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+        release: {{ .Release.Name }}
         role: agent
     spec:
       containers:
         - name: {{ .Values.agent.name }}
-          image: "{{ .Values.agent.image.repository }}:{{ .Values.agent.image.tag }}"
+          image: "{{ .Values.agent.image.repository }}:v{{ .Chart.AppVersion }}"
```

Comment:
` .Chart.AppVersion` will block the user from specifying any other GoCD version.
As helm does not let you override chart constants.

Can we specify `.Values.app.version` and set the default value to `.Chart.AppVersion`?